exchanges/okx/okx_type_convert.go

```diff
@@ -212,35 +210,21 @@ func (a *OrderDetail) UnmarshalJSON(data []byte) error {
 }
 
 // UnmarshalJSON deserializes JSON, and timestamp information.
-func (a *PendingOrderItem) UnmarshalJSON(data []byte) error {
+func (o *PendingOrderItem) UnmarshalJSON(data []byte) error {
```

Comment:
Can remove this method and it was obfuscating an issue with unmarshalling reduce only string. 
```diff
diff --git a/exchanges/okx/okx_type_convert.go b/exchanges/okx/okx_type_convert.go
index a937352a7..6b4e9afc3 100644
--- a/exchanges/okx/okx_type_convert.go
+++ b/exchanges/okx/okx_type_convert.go
@@ -209,24 +209,6 @@ func (a *OrderDetail) UnmarshalJSON(data []byte) error {
        return nil
 }

-// UnmarshalJSON deserializes JSON, and timestamp information.
-func (o *PendingOrderItem) UnmarshalJSON(data []byte) error {
-       type Alias PendingOrderItem
-       a := &struct {
-               *Alias
-               UpdateTime   okxUnixMilliTime `json:"uTime"`
-               CreationTime okxUnixMilliTime `json:"cTime"`
-               FillTime     okxUnixMilliTime `json:"fillTime"`
-       }{
-               Alias: (*Alias)(o),
-       }
-       err := json.Unmarshal(data, a)
-       o.CreationTime = a.CreationTime.Time()
-       o.UpdateTime = a.UpdateTime.Time()
-       o.FillTime = a.FillTime.Time()
-       return err
-}
-
 // UnmarshalJSON deserializes JSON, and timestamp information.
 func (a *RfqTradeResponse) UnmarshalJSON(data []byte) error {
        type Alias RfqTradeResponse
diff --git a/exchanges/okx/okx_types.go b/exchanges/okx/okx_types.go
index 559635069..be2adbe27 100644
--- a/exchanges/okx/okx_types.go
+++ b/exchanges/okx/okx_types.go
@@ -719,7 +719,7 @@ type OrderHistoryRequestParams struct {
 type PendingOrderItem struct {
        AccumulatedFillSize        convert.StringToFloat64 `json:"accFillSz"`
        AveragePrice               convert.StringToFloat64 `json:"avgPx"`
-       CreationTime               time.Time               `json:"cTime"`
+       CreationTime               okxUnixMilliTime        `json:"cTime"`
        Category                   string                  `json:"category"`
        Currency                   string                  `json:"ccy"`
        ClientOrderID              string                  `json:"clOrdId"`
@@ -727,7 +727,7 @@ type PendingOrderItem struct {
        FeeCurrency                currency.Code           `json:"feeCcy"`
        LastFilledPrice            convert.StringToFloat64 `json:"fillPx"`
        LastFilledSize             convert.StringToFloat64 `json:"fillSz"`
-       FillTime                   time.Time               `json:"fillTime"`
+       FillTime                   okxUnixMilliTime        `json:"fillTime"`
        InstrumentID               string                  `json:"instId"`
        InstrumentType             string                  `json:"instType"`
        Leverage                   convert.StringToFloat64 `json:"lever"`
@@ -752,7 +752,7 @@ type PendingOrderItem struct {
        TakeProfitTriggerPrice     convert.StringToFloat64 `json:"tpTriggerPx"`
        TakeProfitTriggerPriceType string                  `json:"tpTriggerPxType"`
        TradeID                    string                  `json:"tradeId"`
-       UpdateTime                 time.Time               `json:"uTime"`
+       UpdateTime                 okxUnixMilliTime        `json:"uTime"`
 }

 // TransactionDetailRequestParams retrieve recently-filled transaction details in the last 3 day.
@@ -2499,7 +2499,7 @@ type WsBalanceAndPosition struct {

 // WsOrder represents a websocket order.
 type WsOrder struct {
-       PendingOrderItem
+       *PendingOrderItem
        AmendResult     string                  `json:"amendResult"`
        Code            string                  `json:"code"`
        ExecType        string                  `json:"execType"`
@@ -2508,7 +2508,7 @@ type WsOrder struct {
        FillNotionalUsd convert.StringToFloat64 `json:"fillNotionalUsd"`
        Msg             string                  `json:"msg"`
        NotionalUSD     convert.StringToFloat64 `json:"notionalUsd"`
-       ReduceOnly      bool                    `json:"reduceOnly"`
+       ReduceOnly      bool                    `json:"reduceOnly,string"`
        RequestID       string                  `json:"reqId"`
 }

diff --git a/exchanges/okx/okx_websocket.go b/exchanges/okx/okx_websocket.go
index 102f5d278..77f0f2312 100644
--- a/exchanges/okx/okx_websocket.go
+++ b/exchanges/okx/okx_websocket.go
@@ -1060,7 +1060,7 @@ func (ok *Okx) wsProcessOrders(respRaw []byte) error {
                        AssetType:            a,
                        AverageExecutedPrice: response.Data[x].AveragePrice.Float64(),
                        ClientOrderID:        response.Data[x].ClientOrderID,
-                       Date:                 response.Data[x].CreationTime,
+                       Date:                 response.Data[x].CreationTime.Time(),
                        Exchange:             ok.Name,
                        ExecutedAmount:       execAmount,
                        RemainingAmount:      remAmount,
@@ -1074,7 +1074,7 @@ func (ok *Okx) wsProcessOrders(respRaw []byte) error {
                        Type:                 orderType,
                }
                if orderStatus == order.Filled {
-                       d.CloseTime = response.Data[x].FillTime
+                       d.CloseTime = response.Data[x].FillTime.Time()
                        if d.Amount == 0 {
                                d.Amount = d.ExecutedAmount
                        }
```

Reply:
That pointer to `PendingOrderItem` can be removed. Whooops.

Reply:
Okay. That's done.
ReduceOnly is part of non-SPOT, so I wasn't (yet) testing it.
I'll get to that soon.
In the meantime I've taken your changes.